### DISCUSSION TITLE
Example of pyvista working in google colab with interactive plot

### DISCUSSION THREAD

**Discussion by thomwolf** [pyvista/pyvista#3369] [Q&A]
I've been struggling to get a working example of pyvista working in a Google Colab.

I've followed the advises here:
- https://github.com/pyvista/pyvista/issues/313
- https://dev.pyvista.org/user-guide/jupyter/index.html

Does anyone has a working example of pyvista diplaying some mesh on Google colab in an interactive setting?

**Comment by banesullivan** [comment]
AFAIK, this should "just work" these days though I am not a Colab user...

AFAIK, the steps are to enable the headless display (using xvfb for example) and making sure you have one of the notebook rendering backends set up: https://docs.pyvista.org/user-guide/jupyter/index.html#usage-with-pyvista

All of this is outlined in the links you shared, so if this isn't working for you, then I suspect something is awry. Could you provide a public Colab notebook for us to debug this?

**Reply by malflink** [reply]
Hi, I work with gempy and gemgis. I have the identical mesh problem. I also use Google colab. Whenever mesh is tried, it simply crashes! I would really like to resolve this

**Reply by banesullivan** [reply]
@giswqs, you use PyVista on Google Colab sometimes, right? Do you know the set up process here?

**Reply by giswqs** [reply]
I could not get pyvista to work on Colab. It always crashes. ChatGPT can't figure it out either. 
https://chatgpt.com/share/67c12cb7-c650-800c-a8a4-dabe7dcfe0ba

**Comment by malflink** [comment]
Hi
This works!
this
mike

**Comment by th-rtyf-re** [comment]
The following worked for me:
```
!pip install pyvista[jupyter]
```
and then
```
import pyvista as pv

pv.set_jupyter_backend('html')
pv.global_theme.notebook = True # otherwise nothing plots
```
